pyfritzhome/devicetypes/fritzhomeentitybase.py

```diff
@@ -42,6 +42,12 @@ def _update_from_node(self, node):
         self.ain = node.attrib["identifier"]
         self._functionsbitmask = int(node.attrib["functionbitmask"])
 
+        # Workaround for broken AVM FRITZ!DECT 440 in FRITZ!OS < 7.20
+        if (node.attrib["manufacturer"] == "AVM"
+            and node.attrib["productname"] == "FRITZ!DECT 440"
+            and node.find("humidity") is None):
+            self._functionsbitmask &= ~int(FritzhomeDeviceFeatures.HUMIDITY)
+
```

Comment:
```suggestion
        # Workaround for broken AVM FRITZ!DECT 440 in FRITZ!OS < 7.24
        if (self._has_feature(FritzhomeDeviceFeatures.HUMIDITY)
            and node.find("humidity") is None):
            self._functionsbitmask &= ~int(FritzhomeDeviceFeatures.HUMIDITY)
```

we should never rely on product or manufacturer names, they can change (_see the latest issues with SMA devices after fritzbox update which changed the device names of all dect products_)
Please test if this suggestions works for you